File: _API_TESTING_GHERKIN_/DEMOAPP001_TYPESCRIPT_CYPRESS/cypress/support/step_definitions/healthcheck/alive_endpoint.steps.ts
import { Given, When, Then } from "@badeball/cypress-cucumber-preprocessor";
import { expect } from "chai";
import { SendGetRequest } from "../../../../screenplay/tasks/SendGetRequest";
import { ResponseStatus } from "../../../../screenplay/questions/ResponseStatus";
import { ResponseBody } from "../../../../screenplay/questions/ResponseBody";
import { apiActor } from "../../../../screenplay/core/api-world";

Given("the API is available", () => {
  return apiActor().attemptsTo(SendGetRequest.to("/alive"));
});

When("a GET request is made to the Alive Endpoint", () => {
  return apiActor().attemptsTo(SendGetRequest.to("/alive"));
});

Then("the API response should return a status code of 200", () => {
  const status = apiActor().answer(ResponseStatus.code());
  expect(status).to.equal(200);
});

Then('the Health Check response body should contain "Status" with the value "ALIVE-AND-KICKING"', () => {
  const body = apiActor().answer(ResponseBody.json());
  expect(body.Status).to.equal("ALIVE-AND-KICKING");
});

File: _API_TESTING_GHERKIN_/DEMOAPP002_CSHARP_PLAYWRIGHT/TokenParserTests/Steps/AliveEndpoint_Steps.cs
using NUnit.Framework.Legacy;
using TechTalk.SpecFlow;
using TokenParserTests.Screenplay;
using TokenParserTests.Screenplay.Questions;
using TokenParserTests.Screenplay.Support;
using TokenParserTests.Screenplay.Tasks;

namespace TokenParserTests.Steps;

[Binding]
public class AliveEndpoint_Steps
{
    private readonly ScenarioContext _scenarioContext;

    public AliveEndpoint_Steps(ScenarioContext scenarioContext)
    {
        _scenarioContext = scenarioContext;
    }

    private Actor Actor => _scenarioContext.GetActor();

    [Given(@"the API is available")]
    public void GiventheAPIIsAvailable()
    {
        // Kept for readability / parity with other stacks
        Console.WriteLine("Given the API is available");
    }

    [When(@"a GET request is made to the Alive Endpoint")]
    public async Task WhenISendAGETRequestTo()
    {
        await Actor.AttemptsTo(SendGetRequest.To("/alive"));
    }

    [Then(@"the API response should return a status code of (.*)")]
    public async Task ThenTheResponseShouldReturnAStatusCodeOf(int statusCode)
    {
        var actualStatus = await Actor.Answer(ResponseStatus.Code());
        Assert.That((int)actualStatus, Is.EqualTo(statusCode));
    }

    [Then(@"the Alive Endpoint response body should contain ""(.*)"" with the value ""(.*)""")]
    public async Task ThenTheResponseShouldContainTheMessage(string propertyName, string expectedString)
    {
        using var json = await Actor.Answer(ResponseJson.Body());
        if (!json.RootElement.TryGetProperty(propertyName, out var actual))
        {
            Assert.Fail($"Response body does not contain '{propertyName}'.");
        }

        Assert.That(actual.GetString(), Is.EqualTo(expectedString),
            $"Expected string: {expectedString}, but got: {actual.GetString()}");
    }
}
